### PR TITLE
check if phpstorm meta path is a file or directory

### PR DESCRIPTION
**Pull request by SignpostMarv**
* swaps `file_exists()` for `is_file()`
* adds a debug note in case someone wonders why their phpstorm meta path isn't being handled

leaving this in draft for now, as I'm wondering if it's worth dropping a `.gitkeep` file in an appropriately named folder for coverage purposes?

### CODE REVIEW

File: src/Psalm/Config.php [hunk 1]
```diff
@@ -1560,8 +1560,10 @@ public function visitStubFiles(Codebase $codebase, Progress $progress = null)
 
         $phpstorm_meta_path = $this->base_dir . DIRECTORY_SEPARATOR . '.phpstorm.meta.php';
 
-        if (file_exists($phpstorm_meta_path)) {
+        if (is_file($phpstorm_meta_path)) {
             $stub_files[] = $phpstorm_meta_path;
+        } elseif (is_directory($phpstorm_meta_path)) {
+            $progress->debug('phpstorm meta path not included in stub files list as it is a directory' . "\n");
```
Comment (weirdan) [inline src/Psalm/Config.php]: The docs say files in this folder have the same format, so they could all be added to `$stub_files` perhaps?

Comment (SignpostMarv) [reply src/Psalm/Config.php]: 🤷‍♂️ *does not use phpstorm*

Comment (SignpostMarv) [reply src/Psalm/Config.php]: How deep does the folder go?

Comment (weirdan) [reply src/Psalm/Config.php]: I'm not a PHPStorm user either. I suppose *all* files should be read, regardless of the folder depth.

Comment (SignpostMarv) [reply src/Psalm/Config.php]: @weirdan https://confluence.jetbrains.com/display/PhpStorm/PhpStorm+Advanced+Metadata

> 2016.2 + -> you can have multiple separate files in FOLDER named `.phpstorm.meta.php`

**Comment by muglug** [comment]
> Maybe it makes more sense to include all files from that directory as stub_file instead of skipping them.

Yeah 👍 on this idea, otherwise this is great.

**Comment by muglug** [comment]
Oh, then why is it still a draft?

**Comment by SignpostMarv** [comment]
> Oh, then why is it still a draft?

I think github might've lost the review comment; wasn't sure if doing the [`realpath()` conditional](https://github.com/vimeo/psalm/commit/7ad5f184a4942137b41018e50aa4d57b25f1d2aa#diff-20a62f59764c9aa89c920a98d1b6ec16R1571) was overly-cautious?

**Comment by muglug** [comment]
> wasn't sure if doing the realpath() conditional was overly-cautious?

I'm fine with that

**Pull request merged**